src/components/CardsListToggle.js

```diff
@@ -35,24 +35,18 @@ const viewToggleButtons = (listView, setListView) => {
   ])
 }
 
-
 Utils.syncAtomToSessionStorage(toggleStateAtom, 'toggleState')
 
-
-const togglesListView = key => {
-  return compose(
-    Utils.connectAtom(toggleStateAtom, 'toggleState'),
-    mapProps(({ toggleState, ...props }) => {
-      const listView = toggleState && toggleState[key]
-
-      return {
-        listView,
-        viewToggleButtons: viewToggleButtons(listView, v => toggleStateAtom.update(m => ({ ...m, [key]: v }))),
-        ...props
-      }
-    })
-  )
+export const useViewToggle = key => {
+  const toggleState = Utils.useAtom(toggleStateAtom)
+  return [toggleState[key], v => toggleStateAtom.update(_.set(key, v))]
 }
 
-
-export default togglesListView
+export const withViewToggle = key => WrappedComponent => {
+  const Wrapper = props => {
+    const [listView, setListView] = useViewToggle(key)
+    return h(WrappedComponent, { ...props, listView, setListView })
```

Comment:
TOL: I am curious about the rationale for opting to make the toggle buttons an explicit export/import vs being added to the props. Is there a benefit of doing it this way? I view the toggle button component being passed as a prop as being slightly simpler, as the developer doesn't have to pass back the two props that the toggle buttons already have access to when defined.

Reply:
It's a fair question. There's no immediate benefit in this case, functionality wise. There are some hypothetical use cases that are easier with this pattern, such as using toggle buttons that aren't tied to session storage, or adding additional styles overrides on the buttons. But mostly it's about keeping the usage patterns consistent with other places. This way the hooks just provide pure data and callbacks (a-la `useState`), and the consumer wires things up as desired. Especially since the interface is such a straightforward kind of `value`/`onChange` pair, it seemed like a natural way to do things, even if slightly more verbose.

Reply:
Thanks for the explanation!